x-pack/solutions/security/plugins/security_solution/public/explore/hosts/pages/details/index.tsx

```diff
@@ -130,7 +134,26 @@ const HostDetailsComponent: React.FC<HostDetailsProps> = ({ detailName, hostDeta
     [dispatch]
   );
 
-  const { indicesExist, selectedPatterns, sourcererDataView } = useSourcererDataView();
+  const {
+    indicesExist: oldIndicesExist,
+    selectedPatterns: oldSelectedPatterns,
+    sourcererDataView: oldSourcererDataView,
+  } = useSourcererDataView();
+
+  const newDataViewPickerEnabled = useIsExperimentalFeatureEnabled('newDataViewPickerEnabled');
+
+  const { dataView } = useDataView();
+  const { dataViewSpec } = useDataViewSpec();
+  const experimentalSelectedPatterns = useSelectedPatterns();
+
+  const sourcererDataView = newDataViewPickerEnabled ? dataViewSpec : oldSourcererDataView;
+  const indicesExist = newDataViewPickerEnabled
+    ? !!dataView?.matchedIndices?.length
```

Comment:
the placeholder page flashes when switching data views. perhaps add a loading check somewhere?

More obvious with some throttling:

https://github.com/user-attachments/assets/35462a43-fd6c-4615-9c76-e33b41bf7a5a

Reply:
hey, we should address this but I dont want to introduce new mechanism here, since it would have to be applied to several other spots. lets discuss somewhere else:)

Reply:
nice catch @christineweng !